moveit_ros/moveit_servo/src/utils/common.cpp

```diff
@@ -119,15 +119,38 @@ trajectory_msgs::msg::JointTrajectory composeTrajectoryMessage(const servo::Para
   joint_trajectory.header.frame_id = servo_params.planning_frame;
   joint_trajectory.joint_names = joint_state.joint_names;
 
-  trajectory_msgs::msg::JointTrajectoryPoint point;
+  static trajectory_msgs::msg::JointTrajectoryPoint point;
   point.time_from_start = rclcpp::Duration::from_seconds(servo_params.publish_period);
+  const size_t num_joints = joint_state.joint_names.size();
+  point.positions.resize(num_joints);
+  point.velocities.resize(num_joints);
+  point.accelerations.resize(num_joints);
 
   // Set the fields of trajectory point based on which fields are requested.
   // Some controllers check that acceleration data is non-empty, even if accelerations are not used
   // Send all zeros (joint_state.accelerations is a vector of all zeros).
-  point.positions = servo_params.publish_joint_positions ? joint_state.positions : point.positions;
-  point.velocities = servo_params.publish_joint_velocities ? joint_state.velocities : point.velocities;
-  point.accelerations = servo_params.publish_joint_accelerations ? joint_state.accelerations : point.accelerations;
+
+  if (servo_params.publish_joint_positions)
+  {
+    for (size_t i = 0; i < num_joints; i++)
+    {
+      point.positions[i] = joint_state.positions[i];
+    }
+  }
+  if (servo_params.publish_joint_velocities)
+  {
+    for (size_t i = 0; i < num_joints; i++)
+    {
+      point.velocities[i] = joint_state.velocities[i];
+    }
+  }
+  if (servo_params.publish_joint_accelerations)
+  {
+    for (size_t i = 0; i < num_joints; i++)
+    {
+      point.accelerations[i] = joint_state.accelerations[i];
+    }
+  }
```

Comment:
Can reduce the number of for-loops here:

```suggestion
    for (size_t i = 0; i < num_joints; i++)
    {
      if (servo_params.publish_joint_positions)
      {
        point.positions[i] = joint_state.positions[i];
      }
      if (servo_params.publish_joint_velocities)
      {
        point.velocities[i] = joint_state.velocities[i];
      }
      if (servo_params.publish_joint_accelerations)
      {
        point.accelerations[i] = joint_state.accelerations[i];
      }
    }
```

Reply:
I had considered putting  them in a single loop ,but from a performance point , I think it is better when we don't have conditionals inside the loop,